src/stdlib/src/lib.rs

```diff
@@ -396,8 +400,19 @@ pub extern "C" fn srandom(seed: c_uint) {
 }
 
 #[no_mangle]
-pub extern "C" fn strtod(s: *const c_char, endptr: *mut *mut c_char) -> c_double {
-    unimplemented!();
+pub unsafe extern "C" fn strtod(s: *const c_char, endptr: *mut *mut c_char) -> c_double {
+    //TODO: endptr
+
+    use core::str::FromStr;
+
+    let s_str = str::from_utf8_unchecked(platform::c_str(s));
```

Comment:
This will end up reading `usize::MAX` bytes if the input string doesn't have a null byte.  I dunno how much we care about this though as the input is supposed to be a C string.

Reply:
Unfortunately, that is exactly what it is supposed to do. Not going through a potentially usize::MAX string of whitespace would not implement the function per the POSIX specification

Reply:
That doesn't sound right to me.  That would imply that you need to loop through the entire string to verify that it is indeed a C string before converting the number.  If that were the case, the following code would not work:
```c
char bytes[] = { ' ', ' ', '4', 'a' };
double result = strtod(bytes, NULL);
```

Reply:
Ah, I see what you mean.